example3/Example/Library.cs

```diff
@@ -0,0 +1,85 @@
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Diagnostics.Metric;
+
+namespace MyLibrary
+{
+    public class Library
+    {
+
+        static Counter s_c1 = new Counter("MyLibrary", "1.0.0", "c1", new string[] { "Label1", "Label2" });

```

Comment:
At the moment I'm a fan of the idea that counters are primarily distinguished by their name and that if library_name or version exist at all, they are optional. A few reasons:
1. It is simple
2. It aligns with what other APIs already do (Prometheus, Micrometer, ILogger)
3. I've never seen particularly good guidance from OpenTelemetry about how developers are expected to set these fields (in particular version) and what value they are getting in return for adding this extra complexity to their code.

Reply:
In tomorrow's Metrics API/SDK SIG meeting I'll bring the version question.
My understanding is that [OpenTelemetry tracer version](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/api.md#get-a-tracer) is used to describe the instrumentation library version, which is controlled by the semantic version of the semantic convention (I know it is very mouthful).

Reply:
And in metrics we (I mean OpenTelemetry community) have intention to keep the metrics provider API consistent with tracer provider API.

Reply:
I'm not opposed to the data existing, put I am questioning making it a required constructor parameter in the .NET API. For example these values are not required inputs to LoggerFactory creation so there has to be some path where OT can consume data when the developer hasn't provided this data.

Reply:
I agree, I don't think putting a version string in the API is a common practice in .NET. I've seen it in some Java IoC (Inversion of Control) framework.
I think OpenTelemetry MeterProvider will have the version info, just to keep it consistent with TracerProvider API.
For .NET, I think we don't add it in the BCL libraries, we use convention instead (similar like `ActivitySource("MyCompany.MyService.MyComponent")`).

Reply:
Version is an optional component per current Spec.  Also, we will want to clarify if version plays any part in the "identity" of a metric.

Reply:
@noahfalk, you took out library name/ver in new Counter().  However, we need to find a way to make that available as OTel currently requires it.

Reply:
It is inside MetricSource now